app/actions/GetJourneyActionRefiner.scala

```diff
@@ -16,29 +16,36 @@
 
 package actions
 
-import play.api.mvc.{ActionRefiner, Request, Result}
+import controllers.support.RequestSupport.hc
+import essttp.journey.JourneyConnector
+import essttp.journey.model.Journey
+import play.api.Logger
+import play.api.mvc.{ActionRefiner, Request, Result, Results}
 import requests.JourneyRequest
-import services.JourneyService
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class GetJourneyActionRefiner @Inject() (
-    journeyService: JourneyService
-)(
+class GetJourneyActionRefiner @Inject() (journeyConnector: JourneyConnector)(
     implicit
     ec: ExecutionContext
 ) extends ActionRefiner[Request, JourneyRequest] {
 
+  private val logger = Logger(getClass)
+
   override protected def refine[A](request: Request[A]): Future[Either[Result, JourneyRequest[A]]] = {
     implicit val r: Request[A] = request
     for {
-      journey <- journeyService.get()
-    } yield Right(new JourneyRequest(journey, request))
+      maybeJourney: Option[Journey] <- journeyConnector.findLatestJourneyBySessionId()
+    } yield maybeJourney match {
+      case Some(journey) => Right(new JourneyRequest(journey, request))
+      case None =>
+        logger.warn(s"No journey found for sessionId: [ ${hc.sessionId} ]")
+        Left(Results.ImATeapot) //todo what should this be, obviously not teapot error
```

Comment:
Maybe we should display content: "For you security we timed you out?"  Let's analyse it.